### PR TITLE
[SPARK-35415][SQL] Change information to struct type for SHOW TABLE EXTENDED command

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Change information column to struct type for SHOW TABLE EXTENDED command.

### Why are the changes needed?
Usually not all information is what we need and it has poor readability. After SPARK-35283 and this PR. We can get the need information by key:

```
WITH s AS (SHOW TABLE EXTENDED LIKE '*') SELECT tableName, information.storage.Provider FROM s
+------------+---------------------+
|tableName   |information[Provider]|
+------------+---------------------+
|test_delta  |delta                |
|test_parquet|parquet              |
+------------+---------------------+
```

### Does this PR introduce any user-facing change?
Yes. The information column type changed.

### How was this patch tested?
Unit test.

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
Make this pr based on @wangyum 's work and invite since https://github.com/apache/spark/pull/32563#discussion_r646357632

**Comment by AngersZhuuuu** [comment]
Ping @cloud-fan

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -48,8 +48,8 @@ license: |
...
+
+  - In Spark 3.2, the output schema of `SHOW TABLE EXTENDED` becomes `namespace: string, tableName: string, isTemporary: boolean, information:struct<Database:string,Table:string,Owner:string,Created Time:date,Last Access:date,Created By:string,Type:string,Provider:string,Bucket:struct<Num Buckets:string,Bucket Columns:string,Sort Columns:string>,Comment:string,View Information:struct<View Text:string,View Original Text:string,View Catalog and Namespace:string,View Query Output Columns:string>,Table Properties:string,Statistics:string,Storage:struct<Location:string,Serde Library:string,InputFormat:string,OutputFormat:string,Compressed:string,Storage Properties:string>,Partition Provider:string,Partition Columns:string,Partition Values:string,Partition Parameters:string,Partition Statistics:string,schema:string>`. In Spark 3.1 or earlier, the `namespace` field was named `database` for the builtin catalog, and no change for the v2 catalogs. In Spark 3.1 or earlier, the `information` field was string type. To restore the old schema, you can set `spark.sql.legacy.keepCommandOutputSchema` to `true`.
```
Comment (HyukjinKwon) [inline docs/sql-migration-guide.md:52]: Can we just show the result from df.printSchema in a code block? seems difficult to read.

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md]: > Can we just show the result from df.printSchema in a code block? seems difficult to read.

How about current ?

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

**Pull request closed without merging**